tcl/util.tcl

```diff
@@ -624,3 +623,40 @@ proc qc::glob_recursive {args} {
     # Call glob and return the results
     return [glob {*}$switches {*}$options -- {*}$patterns]
 }
+
+proc qc::qualified_args2unqualified {} {
+    #| Creates an unqualified variable in the caller's stack frame for each qualified argument
+    #| present in the caller but only if the resulting variable would be unambiguous.
+    #| E.G. Caller's args: { table.foo }
+    #|      -> Variable named "foo" is set in caller.
+    #|      Caller's args: { table.foo other.foo }
+    #|      -> No new variables set in caller because "foo" would be ambiguous.
+    set caller_args [info args [dict get [info frame -1] proc]]
+    foreach item [qc::unambiguous {*}$caller_args] {
+        upvar 1 $item value
+        qc::upset 1 [qc::shortname $item] $value
+    }
+}
+
+proc qc::shortname {name} {
```

Comment:
I think we better make the names of these procs less special `arg_shortname` `args_unambiguous` or something similar